drivers/wifi/siwx917/siwx917_wifi.c

```diff
@@ -203,8 +203,10 @@ static int siwx917_scan(const struct device *dev, struct wifi_scan_params *z_sca
 		return -EBUSY;
 	}
 
-	/* FIXME: fill sl_scan_config with values from z_scan_config */
-	sl_scan_config.type = SL_WIFI_SCAN_TYPE_ACTIVE;
+	if (z_scan_config) {
+		sl_scan_config.type = z_scan_config->scan_type;
+	}
```

Comment:
- I believe it's worth to add a comment explaining we are lucky `enum wifi_scan_type` has the same values than the `SL_WIFI_SCAN_TYPE_` definitions.
 - I believe `z_scan_config` can't be `NULL`. Rather than trying to handle all the possible errors made by the caller, I would `__ASSERT(z_scan_config)` then access to `z_scan_config` without conditions.